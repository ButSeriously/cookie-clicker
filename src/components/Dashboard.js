import React from 'react';
import CookiePanel from './CookiePanel';
import StorePanel from './StorePanel';

class Dashboard extends React.Component {
  state = {
    cookiesAmount: 0,
    cookiesPerSecond: 0,
  }

  componentDidMount = () => {
    setInterval(() => {
      this.setState(prevState => ({
        cookiesAmount: prevState.cookiesAmount + prevState.cookiesPerSecond,
      }));
    }, 1000);
  }

  addCookie = () => {
    this.setState(prevState => ({
      cookiesAmount: prevState.cookiesAmount + 1,
    }));
  }

  buildingBought = (cost) => {
    this.setState(prevState => ({
      cookiesAmount: prevState.cookiesAmount - cost,
    }));
  }

  render() {
    return (
      <div className="dashboard">
        <CookiePanel
          cookiesAmount={this.state.cookiesAmount}
          cookiesPerSecond={this.state.cookiesPerSecond}
          addCookie={this.addCookie}
        />
        <StorePanel
          cookiesAmount={this.state.cookiesAmount}
          buildingBought={this.buildingBought}
        />
      </div>
    );
  }
}

export default Dashboard;
