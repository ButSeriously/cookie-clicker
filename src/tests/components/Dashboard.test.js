import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../../components/Dashboard';

test('should render Dashboard', () => {
  const wrapper = shallow(<Dashboard />);
  expect(wrapper).toMatchSnapshot();
});

test('should add cookie', () => {
  const wrapper = shallow(<Dashboard />);
  wrapper.setState({ cookiesAmount: 10 });
  wrapper.find('CookiePanel').prop('addCookie')();
  expect(wrapper.state('cookiesAmount')).toBe(11);
});
