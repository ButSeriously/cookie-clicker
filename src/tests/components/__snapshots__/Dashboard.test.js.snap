// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard 1`] = `
<div
  className="dashboard"
>
  <CookiePanel
    addCookie={[Function]}
    cookiesAmount={0}
    cookiesPerSecond={0}
  />
  <StorePanel />
</div>
`;
